### ISSUE TITLE
'ansible.builtin.systemd_service' is not a valid attribute for a Play

### ISSUE THREAD

**Issue by petermcd** [ansible/ansible-lint#4278]
##### Summary

Please note this was previously reported in May but the issue was closed (#4185).

When running Ansible Lint against a playbook the following message is output.

 ```
'ansible.builtin.systemd_service' is not a valid attribute for a Play
```

To overcome this I have stopped listing the file.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

This is being triggered by pre-commit. This has been reproduced on my localhost (Mac) and on Github Actions

pre-commit config
```
---
repos:
  - repo: https://github.com/adrienverge/yamllint.git
    rev: v1.35.1
    hooks:
      - id: yamllint
  - repo: https://github.com/ansible/ansible-lint.git
    rev: v24.7.0
    hooks:
      - id: ansible-lint
        files: \.(yaml|yml)$
        verbose: true
```

GitHub action

```
name: pre-commit

on:
  push:
    branches:
      - '*'
  pull_request:
    branches:
      - 'main'

jobs:
  pre-commit:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-python@v2
        with:
          python-version: '3.11'
      - uses: pre-commit/action@v2.0.3
```

Errors locally and remotely are identical.

##### STEPS TO REPRODUCE

The following is the playbook, this is a file containing handlers.

```
---
- name: Restart Postfix
  become: true
  ansible.builtin.systemd_service:
    name: postfix
    state: restarted
    enabled: true

- name: Restart SSH
  become: true
  ansible.builtin.systemd_service:
    name: "{{ ssh_name }}"
    state: restarted
    enabled: true
  vars:
    ssh_name: "{{ 'ssh' if ansible_distribution == 'Ubuntu' else 'sshd' }}"

- name: Restart Docker
  become: true
  ansible.builtin.systemd_service:
    name: docker
    state: restarted
    enabled: true

- name: Reboot
  become: true
  ansible.builtin.reboot:
    pre_reboot_delay: 0
```

##### Desired Behaviour

Ansible Lint should not be failing ansible.builtin.systemd_service

##### Actual Behavior

When running sensible lint the following error is output (on both my local MacBook Air and Github Actions using the latest Ubuntu docker image:

```bash
(.venv) (base) peter@Peters-MacBook-Air ansible % git commit -m "merging repo content"
yamllint.................................................................Passed
Ansible-lint.............................................................Failed
- hook id: ansible-lint
- duration: 3.2s
- exit code: 2

INFO     Identified /Users/peter/PycharmProjects/ansible as project root due .git directory.
INFO     Adding '--pre' to ansible-galaxy collection install because we detected one collection being sourced from git.
INFO     Running ansible-galaxy collection install -v --pre -r /Users/peter/PycharmProjects/ansible/requirements.yml
INFO     Set ANSIBLE_LIBRARY=/Users/peter/.cache/ansible-compat/0cb87f/modules:/Users/peter/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/Users/peter/.cache/ansible-compat/0cb87f/collections:/Users/peter/.ansible/collections:/usr/share/ansible/collections:/Users/peter/.cache/ansible-compat/0cb87f/collections:/Users/peter/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/Users/peter/.cache/ansible-compat/0cb87f/roles:roles:/Users/peter/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Loading ignores from .gitignore
INFO     Excluded: .git
INFO     Loading ignores from .idea/.gitignore
INFO     Loading ignores from .venv/.gitignore
INFO     Excluded: .venv/.gitignore
INFO     Excluded: .venv/bin
INFO     Excluded: .venv/lib
INFO     Excluded: .venv/pyvenv.cfg
INFO     Adding '--pre' to ansible-galaxy collection install because we detected one collection being sourced from git.
INFO     Running ansible-galaxy collection install -v --pre -r /Users/peter/PycharmProjects/ansible/requirements.yml
INFO     Set ANSIBLE_LIBRARY=/Users/peter/.cache/ansible-compat/0cb87f/modules:/Users/peter/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/Users/peter/.cache/ansible-compat/0cb87f/collections:/Users/peter/.ansible/collections:/usr/share/ansible/collections:/Users/peter/.cache/ansible-compat/0cb87f/collections:/Users/peter/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/Users/peter/.cache/ansible-compat/0cb87f/roles:roles:/Users/peter/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Executing syntax check on playbook playbooks/handlers.yaml (0.19s)
INFO     Executing syntax check on playbook playbooks/system-maintenance.yaml (0.21s)
WARNING  Listing 1 violation(s) that are fatal
syntax-check[specific]: 'ansible.builtin.systemd_service' is not a valid attribute for a Play
playbooks/handlers.yaml:2:3


                  Rule Violation Summary                   
 count tag                    profile rule associated tags 
     1 syntax-check[specific] min     core, unskippable    

Failed: 1 failure(s), 0 warning(s) on 37 files.
```

**Comment by hoganton** [comment]
I think it might be because the linter thinks that your handlers.yml file is a playbook and not handler/task file. That's probably why you get the syntax error. Because obviously task and playbook files differ a bit from the syntax used inside.
https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_handlers.html


`INFO     Executing syntax check on playbook playbooks/handlers.yaml (0.19s)`

Handlers are usually defined either inside the playbooks under a handler section or as seperate files inside the ansible roles themselves I think. And not as separate files inside the same directory level as playbooks. That might be the problem.

**Comment by Qalthos** [comment]
I don't really have much additional to add to that. The only documented way to reuse handlers is within a role, where the `handlers/` directory contains task files of handlers. Here you have a task file alongside your playbooks, and ansible-lint is rightfully complaining that it is not a valid list of Plays. You need to either move these to the handlers section of an actual playbook or move it to a role structure for lint to understand what is happening.

**Comment by petermcd** [comment]
Ok thanks for the explanation this makes sense.
